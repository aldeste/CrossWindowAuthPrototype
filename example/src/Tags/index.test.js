import * as Tags from "./";
import React from "react";
import renderer from "react-test-renderer";
import { matcher, serializer } from "jest-styled-components";

expect.addSnapshotSerializer(serializer);
expect.extend(matcher);

Object.keys(Tags).forEach(tag => {
  const currentTag = Tags[tag];
  describe(tag + " tag", () => {
    it("Is a react component", () => {
      expect(typeof currentTag).toBe("function");
      expect(<currentTag />.$$typeof.toString()).toBe("Symbol(react.element)");
    });

    it("Is a styled component", () => {
      expect(currentTag.name).toBe("StyledComponent");
    });

    it("Matches earlier configuration", () => {
      expect(renderer.create(<currentTag />)).toMatchStyledComponentsSnapshot();
    });
  });
});
