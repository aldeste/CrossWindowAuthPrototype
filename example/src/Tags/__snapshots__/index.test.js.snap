// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button tag Matches earlier configuration 1`] = `


<currentTag />
`;

exports[`Text tag Matches earlier configuration 1`] = `


<currentTag />
`;

exports[`TextInput tag Matches earlier configuration 1`] = `


<currentTag />
`;

exports[`View tag Matches earlier configuration 1`] = `


<currentTag />
`;
